pkg/agent/apiserver/handlers/ovsflows/handler.go

```diff
@@ -17,23 +17,31 @@ package ovsflows
 import (
 	"encoding/json"
 	"net/http"
+	"strconv"
 
 	"k8s.io/klog"
 
+	"github.com/vmware-tanzu/antrea/pkg/agent/openflow"
 	"github.com/vmware-tanzu/antrea/pkg/agent/querier"
 	"github.com/vmware-tanzu/antrea/pkg/antctl/transform/common"
+	binding "github.com/vmware-tanzu/antrea/pkg/ovs/openflow"
 )
 
 // Response is the response struct of ovsflows command.
 type Response struct {
 	Flow string `json:"flow,omitempty"`
 }
 
-func getAllFlows(aq querier.AgentQuerier) ([]Response, error) {
+func getFlows(aq querier.AgentQuerier, table binding.TableIDType) ([]Response, error) {
 	resps := []Response{}
-	flowStrs, err := aq.GetOfctlClient().DumpFlows()
+	var flowStrs []string
+	var err error
+	if table != binding.TableIDAll {
+		flowStrs, err = aq.GetOfctlClient().DumpTableFlows(uint8(table))
+	} else {
+		flowStrs, err = aq.GetOfctlClient().DumpFlows()
```

Comment:
this doesn't have to be part of this PR, but I think we should have a function like this one in the client (with some more generic filtering) to avoid having too handle too many cases / combinations in the server code:
```
// nil means wildcard
type Filters struct {
    tableID *uint8
    podKey *string
}

func (c *ofclient) DumpFlowsFilters(filters *Filters) {
  // ...
}
```
IMO that would (greatly?) reduce the complexity of the handler

Reply:
I do not feel we should put filters like podKey to OfctlClient() which is pretty dumb and does not understand K8s concepts at all. So, if we want we could add a new layer to implement the logic, but since API handler is the only consumer now, I feel not necessary to add an extra layer.

Reply:
I wasn't clear. By `podKey` I meant `intf.InterfaceName`.

Reply:
So, you mean OfctlClient or openflow.Client?
OfctlClient is just a wrapper of ovs-ofctl and does not understand interface either.
openflow.Client does not know about ofctl, and I feel should not call into ofctl.

Reply:
I forgot we were using `ovs-ofctl` directly here.

Wasn't the long term plan to have all the requests go through `openflow.Client`? It feels weird to me that we have to first query `openflow.Client` for the match key, then `OfctlClient` for the flows. I think you wanted to proxy all requests through `openflow.Client`, but you're waiting for libOpenflow support?

Reply:
Right. But I assume that will take long to happen, as @wenyingd is busy with many other items.